src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -137,7 +141,53 @@ def is_healthy(self) -> bool:
             return True
         return time.time() - self.last_send_time < DISCOVERY_CHECK_THRESHOLD_SEC
 
+    def __publish_resource_diff_services(self, enrichments: List[Enrichment]):
+        for enrich in enrichments:
+            if not enrich.blocks:
+                continue
+
+            block = enrich.blocks[0]
+            if not isinstance(block, KubernetesDiffBlock):
+                continue
+
+            new_resource = block.new_obj
+            if isinstance(new_resource, Deployment) \
+                    or isinstance(new_resource, DaemonSet) \
+                    or isinstance(new_resource, StatefulSet) \
+                    or isinstance(new_resource, Job) \
+                    or isinstance(new_resource, Pod):
+                containers = extract_containers(new_resource)
+                volumes = extract_volumes(new_resource)
+                meta = new_resource.metadata
+                container_info = [ContainerInfo.get_container_info(container) for container in
+                                  containers] if containers else []
+                volumes_info = [VolumeInfo.get_volume_info(volume) for volume in volumes] if volumes else []
+                config = ServiceConfig(labels=meta.labels or {}, containers=container_info,
+                                       volumes=volumes_info)
+                ready_pods = extract_total_pods(new_resource)
+                total_pods = extract_ready_pods(new_resource)
+
+                services = ServiceInfo(
+                    name=meta.name,
+                    namespace=meta.namespace,
+                    service_type=new_resource.kind,
+                    service_config=config,
+                    ready_pods=ready_pods,
+                    total_pods=total_pods,
+                )
+                self.__publish_new_services([services])
```

Comment:
This line deletes all the existing services, and replace it only with the service that was updated now